sdk/core/Azure.Core/src/DynamicData/MutableJsonChange.cs

```diff
@@ -47,18 +31,63 @@ internal struct MutableJsonChange
 
         public MutableJsonChangeKind ChangeKind { get; }
 
-        public JsonValueKind ValueKind => GetSerializedValue().ValueKind;
+        public readonly JsonValueKind ValueKind => Value switch
+        {
+            null => JsonValueKind.Null,
+            bool b => b ? JsonValueKind.True : JsonValueKind.False,
+            string => JsonValueKind.String,
+            DateTime => JsonValueKind.String,
+            DateTimeOffset => JsonValueKind.String,
+            Guid => JsonValueKind.String,
+            byte => JsonValueKind.Number,
+            sbyte => JsonValueKind.Number,
+            short => JsonValueKind.Number,
+            ushort => JsonValueKind.Number,
+            int => JsonValueKind.Number,
+            uint => JsonValueKind.Number,
+            long => JsonValueKind.Number,
+            ulong => JsonValueKind.Number,
+            float => JsonValueKind.Number,
+            double => JsonValueKind.Number,
+            decimal => JsonValueKind.Number,
+            JsonElement e => e.ValueKind,
+            _ => throw new InvalidOperationException($"Unrecognized change type '{Value.GetType()}'.")
+        };
+
+        internal readonly void EnsureString()
```

Comment:
Could you share an example of how you imagine the call site would look?  My goal with making them kind specific was to make it easy to spot-check from the TryGetXx methods that we were calling the right type - e.g. a DateTime is a string, but a  Single is a number.

Reply:
I assume it would be just as easy to see
`EnsureString()`

as it would be to see
`EnsureIsValueKind(JsonValueKind.String)`

Both require you to make the same connection between type and json payload type. But, I don't feel strongly at all.